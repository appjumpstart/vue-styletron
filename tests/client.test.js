import { shallowMount, createLocalVue } from '@vue/test-utils'

import VueStyletron from '../'
import HelloMessage from './fixtures/HelloMessage.vue'

// Create test-utils-based local Vue instance.
const localVue = createLocalVue()

// Use the VueStyletron Client plugin.
localVue.use(VueStyletron)

test('client generates styles and classes', () => {
  // Create and mount the app.
  const wrapper = shallowMount(HelloMessage, { localVue })

  // The head should contain a styles block.
  expect(document.head).toMatchSnapshot()

  // The wrapper element should contain the rendered component with generated
  // class names.
  expect(wrapper.element).toMatchSnapshot()
})
