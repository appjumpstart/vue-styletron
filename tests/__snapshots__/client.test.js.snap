// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`client generates styles and classes 1`] = `
<head>
  <style
    media=""
  />
</head>
`;

exports[`client generates styles and classes 2`] = `
<div
  class="ae af mr-5"
>
  Hello!
</div>
`;
